racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -7,21 +7,47 @@ SPDX-License-Identifier: GPL-3.0-or-later -->
 <script setup lang="ts">
 import { useHttp } from '@/plugins/http'
 import { useRacksDBAPI } from '@/composables/RacksDBAPI'
-import { ref, onMounted } from 'vue'
-import SearchBar from '@/components/SearchBar.vue'
+import { ref, onMounted, computed } from 'vue'
 import type { Ref } from 'vue'
-import type { Datacenter, Infrastructure, Rack } from '@/composables/RacksDBAPI'
 import BreadCrumbs from '@/components/BreadCrumbs.vue'
+import type { Infrastructure, Rack } from '@/composables/RacksDBAPI'
+import { Switch, SwitchGroup, SwitchLabel } from '@headlessui/vue'
+import { BarsArrowDownIcon, BarsArrowUpIcon } from '@heroicons/vue/24/outline'
+import { Dialog, DialogPanel } from '@headlessui/vue'
+import DatacenterListInfrastructures from '@/components/DatacenterListInfrastructures.vue'
 
 const http = useHttp()
 const racksDBAPI = useRacksDBAPI(http)
-const datacenters: Ref<Array<Datacenter>> = ref([])
-const infrastructures: Ref<Array<Infrastructure>> = ref([])
-const datacenterDetails: Ref<Datacenter | undefined> = ref()
-const racks: Ref<Array<Rack>> = ref([])
+let infrastructures: Array<Infrastructure> = []
 const rackDetails: Ref<Array<Rack>> = ref([])
 const blobURL = ref()
 const showImg = ref(false)
+const input = ref('')
+const hideEmpty = ref(false)
+const alphabeticalOrder = ref(true)
+var filteredRacks = computed(() => {
+  let result: Array<Rack> = rackDetails.value
+  for (let index = 0; index < result.length; index++) {
```

Comment:
The logic is still wrong here, it should just filters an array, please fix it.

Reply:
The for loop was useless, I removed it and just filtered the array.